app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,37 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @post = @user.posts.order(created_at: :desc)
+    @posts = @user.posts.includes(comments: [:author])
   end
 
   def show
-    @user = User.find(params[:user_id])
-    @post = Post.find(params[:id])
+    @post = Post.includes(comments: [:author]).find(params[:id])
+  end
```

Comment:
- [ ] Kindly don't forget that you are supposed to add a screenshot that shows the implementation of how the `N+1` problem is solved when fetching all posts and their comments for a user before and after the fix. ℹ️